README.md
# Light Carousel [![pub package](https://img.shields.io/pub/v/light_carousel.svg)](https://pub.dev/packages/light_carousel)

A Light-Weight Flutter Carousel widget.

## Usage

As simple as using any flutter Widget.

**Example:**
Add the module to your project `pubspec.yaml`:

```yaml
...
dependencies:
 ...
 light_carousel:
...
```

And install it using `flutter packages get` on your project folder. After that, just import the module and use it:

![Widget Carousel Animated](https://raw.githubusercontent.com/aisivan36/light_carousel/0.0.1/screenshot/flutter.png)

````dart
import 'package:light_carousel/light_carousel.dart';

//...
Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          const Center(
            child: SizedBox(
              height: 150.0,
              width: 300.0,
              child: LightCarousel(
                boxFit: BoxFit.cover,
                autoPlay: true,
                animationCurve: Curves.fastOutSlowIn,
                animationDuration: Duration(milliseconds: 1000),
                dotSize: 6.0,
                dotIncreasedColor: Color(0xFFFF335C),
                dotBgColor: Colors.transparent,
                dotPosition: DotPosition.bottomCenter,
                dotVerticalPadding: 10.0,
                showIndicator: true,
                indicatorBgPadding: 7.0,
                images: [
                  NetworkImage(
                      'https://cdn-images-1.medium.com/max/2000/1*GqdzzfB_BHorv7V2NV7Jgg.jpeg'),
                  NetworkImage(
                      'https://cdn-images-1.medium.com/max/2000/1*wnIEgP1gNMrK5gZU7QS0-A.jpeg'),
                  ExactAssetImage("assets/flutter.jpg"),
                ],
              ),
            ),
          ),
          SizedBox(
            height: 450.0,
            width: 350.0,
            child: LightWidget(
              animationDuration: const Duration(milliseconds: 1000),
              autoplay: true,
              animationCurve: Curves.easeInOut,
              boxFit: BoxFit.cover,
              pages: const [
                _Pages(
                    title: 'first page',
                    networkImage:
                        'https://avatars.githubusercontent.com/u/1609975?s=280&v=4'),
                _Pages(title: 'second page', assetsImage: 'assets/flutter.jpg'),
                _Pages(
                    title: 'third page',
                    networkImage:
                        'https://images.ctfassets.net/23aumh6u8s0i/4TsG2mTRrLFhlQ9G1m19sC/4c9f98d56165a0bdd71cbe7b9c2e2484/flutter'),
              ],
           ),
         ),
       ],
     ),
````





![Carousel 1](https://raw.githubusercontent.com/aisivan36/light_carousel/master/screenshots/screenshot01.png)


```dart
import 'package:light_carousel/light_carousel.dart';

//...
SizedBox(
  height: 150.0,
  width: 300.0,
  child: LightCarousel(
    images: [
      NetworkImage('https://cdn-images-1.medium.com/max/2000/1*GqdzzfB_BHorv7V2NV7Jgg.jpeg'),
      NetworkImage('https://cdn-images-1.medium.com/max/2000/1*wnIEgP1gNMrK5gZU7QS0-A.jpeg'),
      ExactAssetImage("assets/flutter.jpg"),
    ],
  )
),

````

![Carousel 2](https://raw.githubusercontent.com/aisivan36/light_carousel/master/screenshots/screenshot02.png)

```dart
import 'package:light_carousel/light_carousel.dart';

//...
SizedBox(
  height: 200.0,
  width: 350.0,
  child: LightCarousel(
    images: [
      NetworkImage('https://cdn-images-1.medium.com/max/2000/1*GqdzzfB_BHorv7V2NV7Jgg.jpeg'),
      NetworkImage('https://cdn-images-1.medium.com/max/2000/1*wnIEgP1gNMrK5gZU7QS0-A.jpeg'),
      ExactAssetImage("assets/flutter.jpg"),
    ],
    dotSize: 4.0,
    dotSpacing: 15.0,
    dotColor: Colors.lightGreenAccent,
    indicatorBgPadding: 5.0,
    dotBgColor: Colors.purple.withOpacity(0.5),
    borderRadius: true,
  )
),

```

![Carousel 3](https://raw.githubusercontent.com/aisivan36/light_carousel/master/screenshots/screenshot03.png)

```dart
import 'package:light_carousel/light_carousel.dart';

//...
SizedBox(
    height: 200.0,
    width: 350.0,
    child: LightCarousel(
      images: [
        NetworkImage('https://cdn-images-1.medium.com/max/2000/1*GqdzzfB_BHorv7V2NV7Jgg.jpeg'),
        NetworkImage('https://cdn-images-1.medium.com/max/2000/1*wnIEgP1gNMrK5gZU7QS0-A.jpeg'),
        ExactAssetImage("assets/flutter.jpg"),
      ],
      dotSize: 4.0,
      dotSpacing: 15.0,
      dotColor: Colors.lightGreenAccent,
      indicatorBgPadding: 5.0,
      dotBgColor: Colors.purple.withOpacity(0.5),
      borderRadius: true,
      moveIndicatorFromBottom: 180.0,
      noRadiusForIndicator: true,
    )
),

```

!![Carousel 4](https://raw.githubusercontent.com/aisivan36/light_carousel/master/screenshots/screenshot04.png)

```dart
import 'package:light_carousel/light_carousel.dart';

//...
SizedBox(
    height: 200.0,
    width: 350.0,
    child: LightCarousel(
      images: [
        NetworkImage('https://cdn-images-1.medium.com/max/2000/1*GqdzzfB_BHorv7V2NV7Jgg.jpeg'),
        NetworkImage('https://cdn-images-1.medium.com/max/2000/1*wnIEgP1gNMrK5gZU7QS0-A.jpeg'),
        ExactAssetImage("assets/flutter.jpg"),
      ],
      dotSize: 4.0,
      dotSpacing: 15.0,
      dotColor: Colors.lightGreenAccent,
      indicatorBgPadding: 5.0,
      dotBgColor: Colors.purple.withOpacity(0.5),
      borderRadius: false,
      moveIndicatorFromBottom: 180.0,
      noRadiusForIndicator: true,
      overlayShadow: true,
      overlayShadowColors: Colors.white,
      overlayShadowSize: 0.7,
    )
),

```

![Carousel 5](https://raw.githubusercontent.com/aisivan36/light_carousel/master/screenshots/screenshot05.png)

```dart
import 'package:light_carousel/light_carousel.dart';

//...
SizedBox(
    height: 200.0,
    width: 350.0,
    child: LightCarousel(
      images: [
        NetworkImage('https://cdn-images-1.medium.com/max/2000/1*GqdzzfB_BHorv7V2NV7Jgg.jpeg'),
        NetworkImage('https://cdn-images-1.medium.com/max/2000/1*wnIEgP1gNMrK5gZU7QS0-A.jpeg'),
        ExactAssetImage("assets/flutter.jpg"),
      ],
      showIndicator: false,
      borderRadius: false,
      moveIndicatorFromBottom: 180.0,
      noRadiusForIndicator: true,
      overlayShadow: true,
      overlayShadowColors: Colors.white,
      overlayShadowSize: 0.7,
    )
),

```


## Parameters and Values

### images

All the images on this Carousel.
**Type:** List

### animationCurve

The transition animation timing curve
**Default value:** Curves.ease
**Type:** Curve

##### Values

- Curves.linear;
- Curves.fastOutSlowIn;
- Curves.ease;
- Curves.bounceOut;
- Curves.bounceIn;
- Curves.bounceInOut;
- Curves.decelerate;
- Curves.ease;
- Curves.easeIn;
- Curves.easeInOut;
- Curves.easeOut;
- Curves.elasticIn;
- Curves.elasticInOut;
- Curves.elasticOut;

### animationDuration

The transition animation duration
**Type:** Duration
**Default value:** 300ms.

### dotSize

The base size of the dots
**Type:** double
**Default value:** 8.0

### dotIncreaseSize

The increase in the size of the selected dot
**Type:** double
**Default value:** 2.0

### dotSpacing

The distance between the center of each dot
**Type:** double
**Default value:** 25.0

### dotColor

The Color of each dot
**Type:** Color
**Default value:** Colors.white

### dotBgColor

The background Color of the dots
**Type:** Color
**Default value:** Colors.grey[800].withOpacity(0.5)

### showIndicator

Enable or Disable the indicator (dots)
**Type:** bool
**Default value:** true

### indicatorBgPadding

Padding Size of the background Indicator
**Type:** double
**Default value:** 20.0

### boxFit

How to show the images in the box
**Type:** BoxFit
**Default value:** cover

##### Values

- BoxFit.cover;
- BoxFit.fitWidth;
- BoxFit.fitHeight;
- BoxFit.scaleDown;
- BoxFit.fill;
- BoxFit.contain;
- BoxFit.none;

### borderRadius

Enable/Disable radius Border for the images
**Type:** bool
**Default value:** false

### radius

Border Radius of the images
**Type:** Radius
**Default value:** Radius.circular(8.0)

### moveIndicatorFromBottom

Move the Indicator From the Bottom
**Type:** double
**Default value:** 0.0

### noRadiusForIndicator

Remove the radius bottom from the indicator background
**Type:** bool
**Default value:** false

### overlayShadow

Enable/Disable Image Overlay Shadow
**Type:** bool
**Default value:** false

### overlayShadowColors

Choose the color of the overlay Shadow color
**Type:** Color
**Default value:** Colors.grey[800]

### overlayShadowSize

Choose the size of the Overlay Shadow, from 0.0 to 1.0
**Type:** double
**Default value:** 0.5

### autoplay

Enable/Disable autoplay carousel
**Type:** bool
**Default value:** true

### autoplayDuration

The autoplay transition duration
**Type:** Duration
**Default value:** 3s

## Credits

This project is made possible by the community surrounding it and especially the wonderful people and projects listed in this document.

- [Ais Ivan](https://github.com/aisivan36)

- [JLouage ](https://github.com/jlouage/flutter-carousel-pro)

## Contributing

Feel free to help!


CHANGELOG.md
## 0.0.4

- Enhanced Documentation and fixed some issues

## 0.0.3

- Fixed

## 0.0.2

- Fixed some issues

## 0.0.1

- Initial Release

* TODO: Describe initial release.


test/light_carousel_test.dart
// import 'package:flutter_test/flutter_test.dart';

// import 'package:light_carousel/light_carousel.dart';

// void main() {
//   test('adds one to input values', () {
//     const calculator = LightCarousel();
//   });
// }


lib/main/widgets.dart
import 'dart:math';
import 'package:flutter/material.dart';
import 'dart:async';

/// Main Light Widget
class LightWidget extends StatefulWidget {
  /// All the pages on this Carousel. Must Provide Widget for Pages
  final List<Widget> pages;

  /// Transtition animation timing curver. The default is [Curves.ease]
  /// `Watch out!` some Curves transition caused stuck failed to animate
  /// Do not use [Curves.easeInBack] and some of unknown animation.
  final Curve animationCurve;

  ///The transition animation duration. Default is 300ms.
  final Duration animationDuration;

  /// The base size of the dots. Default is 8.0
  final double dotSize;

  /// The increase in the size of the selected dot. Default is 2.0
  final double dotIncreaseSize;

  /// The distance between the center of each dot. Default is 25.0
  final double dotSpacing;

  /// The Color of each dot. Default is Colors.white
  final Color? dotColor;

  /// The background Color of the dots. Default is [Colors.grey[800].withOpacity(0.5)]
  final Color? dotBgColor;

  /// Enable or Disable the indicator (dots). Default is true
  final bool showIndicator;

  /// Padding Size of the background Indicator. Default is 20.0
  final double indicatorBgPadding;

  /// How to show the images in the box. Default is cover
  final BoxFit boxFit;

  /// Enable/Disable radius Border for the images. Default is false
  final bool borderRadius;

  /// Border Radius of the images. Default is [Radius.circular(8.0)]
  final Radius? radius;

  /// Move the Indicator From the Bottom
  final double moveIndicatorFromBottom;

  /// Remove the radius bottom from the indicator background. Default false
  final bool noRadiusForIndicator;

  /// Enable/Disable Image Overlay Shadow. Default false
  final bool overlayShadow;

  /// Choose the color of the overlay Shadow color. Default Colors.grey[800]
  final Color? overlayShadowColors;

  /// Choose the size of the Overlay Shadow, from 0.0 to 1.0. Default 0.5
  final double overlayShadowSize;

  /// Enable/Disable the auto play of the slider. Default is [true]
  final bool autoPlay;

  /// [Duration] of the Auto play slider by seconds. Default 3 seconds
  final Duration autoPlayDuration;

  const LightWidget(
      {Key? key,
      required this.pages,
      this.animationCurve = Curves.ease,
      this.animationDuration = const Duration(milliseconds: 300),
      this.dotSize = 8.0,
      this.dotSpacing = 25.0,
      this.dotIncreaseSize = 2.0,
      this.dotColor = Colors.white,
      this.dotBgColor,
      this.showIndicator = true,
      this.indicatorBgPadding = 20.0,
      this.boxFit = BoxFit.cover,
      this.borderRadius = false,
      this.radius,
      this.moveIndicatorFromBottom = 0.0,
      this.noRadiusForIndicator = false,
      this.overlayShadow = false,
      this.overlayShadowColors,
      this.overlayShadowSize = 0.5,
      this.autoPlay = true,
      this.autoPlayDuration = const Duration(seconds: 3)})
      : assert(animationCurve != Curves.easeInBack,
            'Do not use Curves.easeInBack it caused animate failed!'),
        super(key: key);

  @override
  State createState() => LightWidgetState();
}

class LightWidgetState extends State<LightWidget> {
  PageController controller = PageController();
  Timer? timer;

  @override
  void initState() {
    super.initState();

    if (widget.autoPlay) {
      timer = Timer.periodic(widget.autoPlayDuration, (_) {
        if (controller.page == widget.pages.length - 1) {
          controller.animateToPage(
            0,
            duration: widget.animationDuration,
            curve: widget.animationCurve,
          );
        } else {
          controller.nextPage(
            duration: widget.animationDuration,
            curve: widget.animationCurve,
          );
        }
      });
    }
  }

  @override
  void dispose() {
    controller.dispose();
    timer?.cancel();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final List<Widget>? listPages = widget.pages
        .map((widget) => SizedBox(
              child: widget,
            ))
        .toList();

    return Stack(
      children: <Widget>[
        PageView(
          physics: const AlwaysScrollableScrollPhysics(),
          controller: controller,
          children: listPages!,
        ),
        widget.showIndicator
            ? Positioned(
                bottom: widget.moveIndicatorFromBottom,
                left: 0.0,
                right: 0.0,
                child: Container(
                  decoration: BoxDecoration(
                    color:
                        widget.dotBgColor ?? Colors.grey[800]?.withOpacity(0.5),
                    borderRadius: widget.borderRadius
                        ? (widget.noRadiusForIndicator
                            ? null
                            : BorderRadius.only(
                                bottomLeft:
                                    widget.radius ?? const Radius.circular(8.0),
                                bottomRight: widget.radius ??
                                    const Radius.circular(8.0)))
                        : null,
                  ),
                  padding: EdgeInsets.all(widget.indicatorBgPadding),
                  child: Center(
                    child: DotsIndicatorWidget(
                      controller: controller,
                      itemCount: listPages.length,
                      color: widget.dotColor!,
                      dotSize: widget.dotSize,
                      dotSpacing: widget.dotSpacing,
                      dotIncreaseSize: widget.dotIncreaseSize,
                      onPageSelected: (int page) {
                        controller.animateToPage(
                          page,
                          duration: widget.animationDuration,
                          curve: widget.animationCurve,
                        );
                      },
                    ),
                  ),
                ),
              )
            : const SizedBox(),
      ],
    );
  }
}

/// An indicator showing the currently selected page of a PageController
class DotsIndicatorWidget extends AnimatedWidget {
  DotsIndicatorWidget({
    Key? key,
    this.controller,
    this.itemCount,
    this.onPageSelected,
    this.color,
    this.dotSize,
    this.dotIncreaseSize,
    this.dotSpacing,
  }) : super(key: key, listenable: controller ?? PageController());

  /// The [PageController] that this [DotsIndicator] is representing.
  final PageController? controller;

  /// The number of items managed by the [PageController]
  final int? itemCount;

  /// Called when a dot is [tapped]
  final ValueChanged<int>? onPageSelected;

  /// The [color] of the [dots].
  final Color? color;

  /// The base [size] of the dots
  final double? dotSize;

  /// The increase in the [size] of the selected [dot]
  final double? dotIncreaseSize;

  /// The distance between the center of each [dot]
  final double? dotSpacing;

  Widget _buildDot(int index) {
    double selectedness = Curves.easeOut.transform(
      max(
        0.0,
        1.0 - ((controller?.page ?? controller!.initialPage) - index).abs(),
      ),
    );
    double zoom = 1.0 + (dotIncreaseSize! - 1.0) * selectedness;
    return SizedBox(
      width: dotSpacing,
      child: Center(
        child: Material(
          color: color,
          type: MaterialType.circle,
          child: SizedBox(
            width: dotSize! * zoom,
            height: dotSize! * zoom,
            child: InkWell(
              onTap: () => onPageSelected!(index),
            ),
          ),
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Row(
      mainAxisAlignment: MainAxisAlignment.center,
      children: List<Widget>.generate(itemCount!, _buildDot),
    );
  }
}


lib/main/light_widgets.dart
import 'dart:math';
import 'package:flutter/material.dart';
import 'dart:async';

/// Main Light Widget
class LightWidget extends StatefulWidget {
  /// All the pages on this Carousel. Must Provide Widget for Pages
  final List<Widget> pages;

  /// Transtition animation timing curver. The default is [Curves.ease]
  /// `Watch out!` some Curves transition caused stuck failed to animate
  /// Do not use [Curves.easeInBack] and some of unknown animation.
  final Curve animationCurve;

  ///The transition animation duration. Default is 300ms.
  final Duration animationDuration;

  /// The base size of the dots. Default is 8.0
  final double dotSize;

  /// The increase in the size of the selected dot. Default is 2.0
  final double dotIncreaseSize;

  /// The distance between the center of each dot. Default is 25.0
  final double dotSpacing;

  /// The Color of each dot. Default is Colors.white
  final Color dotColor;

  /// The background Color of the dots. Default is [Colors.grey[800].withOpacity(0.5)]
  final Color? dotBgColor;

  /// Enable or Disable the indicator (dots). Default is true
  final bool showIndicator;

  /// Padding Size of the background Indicator. Default is 20.0
  final double indicatorBgPadding;

  /// How to show the images in the box. Default is cover
  final BoxFit boxFit;

  /// Enable/Disable radius Border for the images. Default is false
  final bool borderRadius;

  /// Border Radius of the images. Default is [Radius.circular(8.0)]
  final Radius? radius;

  /// Move the Indicator From the Bottom
  final double moveIndicatorFromBottom;

  /// Remove the radius bottom from the indicator background. Default false
  final bool noRadiusForIndicator;

  /// Enable/Disable Image Overlay Shadow. Default false
  final bool overlayShadow;

  /// Choose the color of the overlay Shadow color. Default Colors.grey[800]
  final Color? overlayShadowColors;

  /// Choose the size of the Overlay Shadow, from 0.0 to 1.0. Default 0.5
  final double overlayShadowSize;

  /// Enable/Disable the auto play of the slider. Default is [true]
  final bool autoPlay;

  /// [Duration] of the Auto play slider by seconds. Default 3 seconds
  final Duration autoPlayDuration;

  const LightWidget(
      {Key? key,
      required this.pages,
      this.animationCurve = Curves.ease,
      this.animationDuration = const Duration(milliseconds: 300),
      this.dotSize = 8.0,
      this.dotSpacing = 25.0,
      this.dotIncreaseSize = 2.0,
      this.dotColor = Colors.white,
      this.dotBgColor,
      this.showIndicator = true,
      this.indicatorBgPadding = 20.0,
      this.boxFit = BoxFit.cover,
      this.borderRadius = false,
      this.radius,
      this.moveIndicatorFromBottom = 0.0,
      this.noRadiusForIndicator = false,
      this.overlayShadow = false,
      this.overlayShadowColors,
      this.overlayShadowSize = 0.5,
      this.autoPlay = true,
      this.autoPlayDuration = const Duration(seconds: 3)})
      : assert(animationCurve != Curves.easeInBack,
            'Do not use Curves.easeInBack it caused animate failed!'),
        super(key: key);

  @override
  State createState() => LightWidgetState();
}

class LightWidgetState extends State<LightWidget> {
  PageController controller = PageController();
  Timer? timer;

  @override
  void initState() {
    super.initState();

    if (widget.autoPlay) {
      timer = Timer.periodic(widget.autoPlayDuration, (_) {
        if (controller.page == widget.pages.length - 1) {
          controller.animateToPage(
            0,
            duration: widget.animationDuration,
            curve: widget.animationCurve,
          );
        } else {
          controller.nextPage(
            duration: widget.animationDuration,
            curve: widget.animationCurve,
          );
        }
      });
    }
  }

  @override
  void dispose() {
    controller.dispose();
    timer?.cancel();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final List<Widget>? listPages = widget.pages
        .map((widget) => SizedBox(
              child: widget,
            ))
        .toList();

    return Stack(
      children: <Widget>[
        PageView(
          physics: const AlwaysScrollableScrollPhysics(),
          controller: controller,
          children: listPages!,
        ),
        widget.showIndicator
            ? Positioned(
                bottom: widget.moveIndicatorFromBottom,
                left: 0.0,
                right: 0.0,
                child: Container(
                  decoration: BoxDecoration(
                    color:
                        widget.dotBgColor ?? Colors.grey[800]?.withOpacity(0.5),
                    borderRadius: widget.borderRadius
                        ? (widget.noRadiusForIndicator
                            ? null
                            : BorderRadius.only(
                                bottomLeft:
                                    widget.radius ?? const Radius.circular(8.0),
                                bottomRight:
                                    widget.radius ?? const Radius.circular(8.0),
                              ))
                        : null,
                  ),
                  padding: EdgeInsets.all(widget.indicatorBgPadding),
                  child: Center(
                    child: DotsIndicatorWidget(
                      controller: controller,
                      itemCount: listPages.length,
                      color: widget.dotColor,
                      dotSize: widget.dotSize,
                      dotSpacing: widget.dotSpacing,
                      dotIncreaseSize: widget.dotIncreaseSize,
                      onPageSelected: (int page) {
                        controller.animateToPage(
                          page,
                          duration: widget.animationDuration,
                          curve: widget.animationCurve,
                        );
                      },
                    ),
                  ),
                ),
              )
            : const SizedBox(),
      ],
    );
  }
}

/// An indicator showing the currently selected page of a PageController
class DotsIndicatorWidget extends AnimatedWidget {
  DotsIndicatorWidget({
    Key? key,
    this.controller,
    this.itemCount,
    this.onPageSelected,
    this.color,
    this.dotSize,
    this.dotIncreaseSize,
    this.dotSpacing,
  }) : super(key: key, listenable: controller ?? PageController());

  /// The [PageController] that this [DotsIndicator] is representing.
  final PageController? controller;

  /// The number of items managed by the [PageController]
  final int? itemCount;

  /// Called when a dot is [tapped]
  final ValueChanged<int>? onPageSelected;

  /// The [color] of the [dots].
  final Color? color;

  /// The base [size] of the dots
  final double? dotSize;

  /// The increase in the [size] of the selected [dot]
  final double? dotIncreaseSize;

  /// The distance between the center of each [dot]
  final double? dotSpacing;

  Widget _buildDot(int index) {
    double selectedness = Curves.easeOut.transform(
      max(
        0.0,
        1.0 - ((controller?.page ?? controller!.initialPage) - index).abs(),
      ),
    );
    double zoom = 1.0 + (dotIncreaseSize! - 1.0) * selectedness;
    return SizedBox(
      width: dotSpacing,
      child: Center(
        child: Material(
          color: color,
          type: MaterialType.circle,
          child: SizedBox(
            width: dotSize! * zoom,
            height: dotSize! * zoom,
            child: InkWell(
              onTap: () => onPageSelected!(index),
            ),
          ),
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Row(
      mainAxisAlignment: MainAxisAlignment.center,
      children: List<Widget>.generate(itemCount!, _buildDot),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:light_carousel/light_carousel.dart';

void main() {
  runApp(
    const MaterialApp(
      debugShowCheckedModeBanner: true,
      title: 'Light Carousel',
      home: CarouselPage(),
    ),
  );
}

class CarouselPage extends StatelessWidget {
  const CarouselPage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Center(
            child: SizedBox(
              height: 150.0,
              width: 300.0,
              child: LightCarousel(
                boxFit: BoxFit.cover,
                autoPlay: true,
                animationCurve: Curves.fastOutSlowIn,
                animationDuration: const Duration(milliseconds: 1000),
                dotSize: 6.0,
                dotIncreasedColor: const Color(0xFFFF335C),
                dotBgColor: Colors.transparent,
                dotPosition: DotPosition.bottomCenter,
                dotVerticalPadding: 10.0,
                showIndicator: true,
                indicatorBgPadding: 7.0,
                images: const [
                  AssetImage('assets/flutter.jpg'),
                  NetworkImage(
                      'https://cdn-images-1.medium.com/max/2000/1*GqdzzfB_BHorv7V2NV7Jgg.jpeg'),
                  NetworkImage(
                      'https://cdn-images-1.medium.com/max/2000/1*wnIEgP1gNMrK5gZU7QS0-A.jpeg'),
                  ExactAssetImage("assets/flutter.jpg"),
                ],
              ),
            ),
          ),
          SizedBox(
            height: 450.0,
            width: 350.0,
            child: LightWidget(
              animationDuration: const Duration(milliseconds: 1000),
              autoPlay: true,
              animationCurve: Curves.easeInOut,
              boxFit: BoxFit.cover,
              pages: const [
                _Pages(
                    title: 'first page',
                    networkImage:
                        'https://avatars.githubusercontent.com/u/1609975?s=280&v=4'),
                _Pages(title: 'second page', assetsImage: 'assets/flutter.jpg'),
                _Pages(
                    title: 'third page',
                    networkImage:
                        'https://images.ctfassets.net/23aumh6u8s0i/4TsG2mTRrLFhlQ9G1m19sC/4c9f98d56165a0bdd71cbe7b9c2e2484/flutter'),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

class _Pages extends StatelessWidget {
  const _Pages({
    Key? key,
    required this.title,
    this.assetsImage,
    this.networkImage,
  })  : assert(assetsImage != null || networkImage != null),
        super(key: key);
  final String title;
  final String? assetsImage, networkImage;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Container(
          padding: const EdgeInsets.only(top: 30),
          height: 50,
          width: 70,
          child: Text(title),
        ),
        Center(
          child: Column(
            children: [
              assetsImage != null
                  ? Image(image: AssetImage(assetsImage!))
                  : const SizedBox(),
              networkImage != null
                  ? Image(image: NetworkImage(networkImage!))
                  : const SizedBox(),
            ],
          ),
        ),
      ],
    );
  }
}
